scripts/portal-flow.js

```diff
@@ -1,7 +1,89 @@
-module.exports = function(runner) {
+'use strict';
+const configureRequest = require('../util/configureRequest');
+const requestBodyUtils = require('../util/sync_request_bodies');
+const recordUtils = require('../util/generate_record');
+const Promise = require('bluebird');
+let baseUrl = '';
+let clientId = '';
+let request;
+
+function urlFor(dataset) {
+  return `${baseUrl}/mbaas/sync/${dataset}`;
+}
+
+function syncDataset(name) {
+  const payload = requestBodyUtils.getSyncRecordsRequestBody({
+    dataset_id: name,
+    meta_data: {
+      clientIdentifier: clientId
+    },
+    pending: []
+  });
+  return request.post({
+    url: urlFor(name),
+    body: payload,
+    json: true
+  });
+}
+
+function create(dataset, data) {
+  const payload = requestBodyUtils.getSyncRecordsRequestBody({
+    fn: 'sync',
+    meta_data: {
+      clientIdentifier: clientId
+    },
+    pending: [recordUtils.generateRecord(data)]
+  });
+  return request.post({
+    url: urlFor('workorders'),
+    body: payload,
+    json: true
+  });
+}
+
+module.exports = function(runner, argv) {
   return function(previousResolution) {
     runner.actStart('Portal Flow');
-    runner.actEnd('Portal Flow');
-    return Promise.resolve(previousResolution);
+    baseUrl = argv.app;
+    clientId = previousResolution.clientIdentifier;
+    request = configureRequest(clientId, previousResolution.sessionToken);
```

Comment:
Ah right. Another way that you could consider (different, but not necessarily better) is currying the functions, so that e.g. `urlFor` would look something like this:

```javascript
// change it to make the function we want to use
function makeUrlFor(baseUrl) {
  return dataset => `${baseUrl}/mbaas/sync/${dataset}`;
}

//Then when we know what the baseUrl is:
const baseUrl = argv.app;
const urlFor = makeUrlFor(baseUrl);
```